### PR TITLE
IOD-115 - Git Action to build/deploy to DEV

### PR DESCRIPTION
**Pull request by akroon3r**
# Description

This PR includes the following proposed change(s):

IOD-115

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation
- [ ] Version change

if your change is a breaking change, please add `breaking change` label to this PR

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

## Does the change impact or break the Docker build?

- [ ] Yes
- [x] No

If Yes: Has Docker been updated accordingly?

- [ ] Yes
- [x] No

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by aideninbcgov — DISMISSED** [review]
I don't want to interrupt your process, but at some point, it'd be very helpful if you could space some time and explain this to us such as how it's done and why.

File: .github/workflows/dev-api-build-deploy.yaml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        run : |
+          docker login ${{ secrets.OPENSHIFT4_EXTERNAL_REPOSITORY }} -u ${{ secrets.OPENSHIFT4_SA_USERNAME }} -p ${{ secrets.OPENSHIFT4_SA_PASSWORD }}
+            
+      - name: Build the Docker image
+        run: |            
+          docker-compose build slam-api
+          
+      - name: Docker Tag Version
+        run: |  
+          docker tag jag-efax_webservice ${{ secrets.OPENSHIFT4_EXTERNAL_REPOSITORY }}/${{ secrets.OPENSHIFT4_TOOLS_NAMESPACE }}/jag-efax-webservice:${{ env.LATEST_VERSION }}
```
Comment (SDToews) [inline .github/workflows/dev-api-build-deploy.yaml:34]: what is jag-efax_webservice?

**Comment by akroon3r** [comment]
> I don't want to interrupt your process, but at some point, it'd be very helpful if you could space some time and explain this to us such as how it's done and why.

Dude, you never have to ask for time to review. You're interrupting nothing. I've scheduled something for us at 9am tomorrow to go over the contents of the PR. Let me know if you need me to move the time.

File: .github/workflows/dev-api-build-deploy.yaml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+      
+      - name: Deploy to DEV
+        env:
+          NAMESPACE: "${{ secrets.OPENSHIFT_DEV_NAMESPACE }}"
+          APPLICATION_NAME: "${{ secrets.APPLICATION_NAME }}"
+          LICENSE_PLATE: "${{ secrets.LICENSE_PLATE }}"
+          ENVIRONMENT: "${{ secrets.ENVIRONMENT }}"
+        run: |  
+          ./.github/helpers/deploy_efax.sh ${{ env.LATEST_VERSION }}
+
```
Comment (SDToews) [inline .github/workflows/dev-api-build-deploy.yaml:48]: don't think deploy_efax.sh exists...

File: openshift/templates/citz-imb-slam-api/dc.yaml [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+      labels:
+        app: ${APPLICATION_NAME}
+        deploymentconfig: ${APPLICATION_NAME}
+        role: api
+    spec:
+      containers:
+        - name: ${APPLICATION_NAME}
+          image: >-
+            image-registry.openshift-image-registry.svc:5000/ca94a8-dev/citz-imb-slam-api-dc@sha256:89b1b50733c6e9c5b97874fcf4e8c7c2669ac3e6092dd2bcdc39ab618724bd37
+          envFrom:
```
Comment (SDToews) [inline openshift/templates/citz-imb-slam-api/dc.yaml:60]: this license plate is hard-coded and the others are not, is that ok?

**Review by SDToews — CHANGES_REQUESTED** [review]
please address specific comments

File: .github/workflows/dev-api-build-deploy.yaml (hunk 1 above)
Comment (akroon3r) [reply .github/workflows/dev-api-build-deploy.yaml:34]: the template I copied this from! good catch, I'm tired at EOD. lmao.

File: .github/workflows/dev-api-build-deploy.yaml (hunk 2 above)
Comment (akroon3r) [reply .github/workflows/dev-api-build-deploy.yaml:48]: good catch!

File: openshift/templates/citz-imb-slam-api/dc.yaml (hunk 3 above)
Comment (akroon3r) [reply openshift/templates/citz-imb-slam-api/dc.yaml:60]: yes this is just a log line in the dc from the previous deploy referencing the image and location

File: .github/workflows/dev-api-build-deploy.yaml (hunk 2 above)
Comment (akroon3r) [reply .github/workflows/dev-api-build-deploy.yaml:48]: see new commit.

File: .github/workflows/dev-api-build-deploy.yaml (hunk 1 above)
Comment (akroon3r) [reply .github/workflows/dev-api-build-deploy.yaml]: see new commit.

**Review by SDToews — APPROVED** [review]

**Pull request merged**